### ISSUE TITLE
Feature Enum Simplification

### ISSUE THREAD

**Issue by dahlb** [dahlb/blueair_api#30]
I have been thinking about the feature enum for a bit. The information is actually somewhat redundant, because the configuration object should have already has this: if a sensor / state does not exist in the get_info() configuration field, then that attribute shall be marked as unsupported.

Thinking along these lines here is a potentially easier-to-maintain solution:

in DeviceAws:

class DeviceAws:

    fan_auto_mode: bool | None | NotImplemented

    async def refresh(self):
         if not find_sensor_in_configuration(info[configuration], 'autofs'):
             self.fan_auto_mode = NotImplemented
         else:
              self.fan_auto_mode = safe_get.....

    async def set_fan_auto_mode(self, value):
           if self.fan_auto_mode is NotImplemented:
                 raise ValueError(".....")

Then in ha_blueair entity scripts,

class AwsFanAutoModeEntity:
    @classmehotd
    def supported(self, device):
        return device.fan_auto_mode is not NotImplemented

# and loop over all entity classes in the list, only add the entity if supported is True.
[rainwoodman](https://github.com/rainwoodman) commented [yesterday](https://github.com/dahlb/blueair_api/commit/9cf7ec9e96a05170cb1fc4e7c18b08e3d140541d#r150084824)
@rainwoodman
rainwoodman
yesterday
Contributor
Also note the current enum is already per attribute -- except filtre_expired, which could be replaced by the filter usage percentage, and handled in the Entity rather than in the coordinator. Actually, the percentage could potentially be visualized with something like a mushroom card to provided more information, and I would recommend deprecating the filter_expired sensor altogether (e.g. the BlueAir app also shows the percentage, not just 'filter exipred').

Another programmatical benefit is to avoid using __new__ override, which while being fun and fulfilling, also is an indicator of deviation from the ordinary use of the Python language constructs: essentially the current pattern is using enum class definition as a configuration language here. For the scope of our project I don't think this is a practical problem, as the amount of configuration is not that big (I suspect only 10s of devices need to be declared and we will at most add a quirks enum field later.

**Comment by dahlb** [comment]
I'm open to this change, it might be more reliable to wait until we can some unit tests to help reduce regressions with this level of a refactor?

**Comment by rainwoodman** [comment]
One of us should update ha_blueair for this. I have some idea how to bake this, but do you have pending changes that shall land first?

**Comment by dahlb** [comment]
I don't have anything else pending, I've paused looking into other issues on the ha_blueair repo until this refactor with the tests and not implemented is finished.  If you'd like to tackle it I am happy to defer. :). I remember you suggesting a method to using the not implemented, and we could switch the model enum back to StrEnum after that too.

**Comment by rainwoodman** [comment]
i missed one important thing. The switches should depend on the existence / implementedness of the setter methods. But how do we best check that? Currently in the update coordinator we say the function raises a NotImplementedError, but that's not checkable at runtime.

**Comment by dahlb** [comment]
1)
I would prefer to do the simple approach of just changing BlueairChildLockSwitchEntity to check 
```
    @classmethod
    def is_implemented(kls, coordinator):
        return coordinator.child_lock is not NotImplemented and isinstance(coordinator, BlueairUpdateCoordinatorDeviceAws)
```
2)
though I think this would work as is, but I agree is feels heavy and has the side affect of calling a set that should have no effect but could cause edge cases
```
        try:
            return coordinator.child_lock is not NotImplemented or coordinator.set_child_lock(coordinator.child_lock)
        except NotImplementedError:
            return False
```

3)
after working through the humidifier pr the abstract method approach seems hard to follow, changing the base implementation to raise NotImplemented instead of being abstract with pass might be more obvious to new comers, thoughts? if we do that we could check for sub class implementations for is_implemented on setters but I'm not sure that's as easy to understand as the code at the top of this comment, even if it would auto reflect improvements made in the coordinator.  I think somebody that is adding functionality will think to update the switch entity's is_implemented

I'm happy with 1 or 3 and could quickly implement, do you have a preference between those two?
